java/simulator/src/main/java/com/hazelcast/simulator/utils/BatchedHistogramLogProcessor.java

```diff
@@ -27,8 +29,18 @@ public static void main(String[] args)
         ExecutorService executor = Executors.newFixedThreadPool(threadCount);
         try {
             List<CompletableFuture<Void>> tasks = new ArrayList<>();
+            Semaphore gate = new Semaphore(1000);
```

Comment:
instead of this, would it be better to use `ThreadPoolExecutor` with a bounded blocking queue

```
new ThreadPoolExecutor(nThreads, nThreads,
                                      0L, TimeUnit.MILLISECONDS,
                                      new LinkedBlockingQueue<Runnable>(1000));
```